servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultHost.java

```diff
@@ -640,4 +648,63 @@ public String toString() {
                     '}';
         }
     }
+
+    private static final class InstrumentedConnectionFactory<Addr, C extends LoadBalancedConnection>
+            extends DelegatingConnectionFactory<Addr, C> {
+
+        private final ConnectTracker connectTracker;
+
+        InstrumentedConnectionFactory(final ConnectionFactory<Addr, C> delegate, ConnectTracker connectTracker) {
+            super(delegate);
+            this.connectTracker = connectTracker;
+        }
+
+        @Override
+        public Single<C> newConnection(Addr addr, @Nullable ContextMap context, @Nullable TransportObserver observer) {
+            return super.newConnection(addr, context, observer).liftSync(delegate -> new ConnectSubscriber(delegate));
```

Comment:
For delegate patter, prefer using `delegate()` instead of `super`. We had cases in the past when `super` implementation changed for backward compatibility and it resulted in unexpected behavior change for users who had filters calling `super` instead of `delegate()`.